### DISCUSSION TITLE
source browser with doxygen(1)

### DISCUSSION THREAD

**Discussion by domo141** [labwc/labwc#1409] [Show and tell]
With the following file (I've named as `doxyfile-x`)

```
# Difference with default Doxyfile 1.9.7
PROJECT_NAME           = "source browser"
OUTPUT_DIRECTORY       = source-browser
TAB_SIZE               = 8
OPTIMIZE_OUTPUT_FOR_C  = YES
EXTRACT_ALL            = YES
EXTRACT_PRIVATE        = YES
EXTRACT_STATIC         = YES
INPUT                  = src include
FILE_PATTERNS          = *.c \
                         *.h
RECURSIVE              = YES
SOURCE_BROWSER         = YES
INLINE_SOURCES         = YES
REFERENCED_BY_RELATION = YES
REFERENCES_RELATION    = YES
HTML_COLORSTYLE        = DARK
GENERATE_LATEX         = NO
HAVE_DOT               = YES
CALL_GRAPH             = YES
CALLER_GRAPH           = YES
```

and then running `doxygen doxyfile-x` in labwc git toplevel directory

Creates `./source-browser/html/...` with 308 megabytes of content. (Outcommenting
`#HAVE_DOT` drops that to 22 MB and runs much faster as it does not create
2000+ png images.)

After a bit of exploration starting at `file://$PWD/source-browser/html/index.html`
one may find this useful.

If that is viewed thru `https://` or `http://` then ensure `*.css` files are served as `text/css`
(otherwise pages are harder to read ;) (`curl -D - https://.../tabs.css` will tell...)

The `doxyfile-x` was created from the `Doxyfile` generated by `doxygen(1)`, and then edited
and finally with `doxygen -x Doxyfile > doxyfile-x`. If one wants to recreate the full Doxyfile,
`doxygen -u doxyfile-x` will do that (overwriting `doxyfile-x`, though).

**Comment by johanmalm** [comment]
Thanks for sharing. Great.
